compute_worker/compute_worker.py

```diff
@@ -362,7 +362,51 @@ def _get_bundle(self, url, destination, cache=True):
 
         # Extract the contents to destination directory
         with ZipFile(bundle_file, 'r') as z:
-            z.extractall(os.path.join(self.root_dir, destination))
+
+            # Run for scoring program only
+            if destination == "program" and self.is_scoring:
+
+                # Check if scoring program is zipped with or without a parent directory
+                # If parent directory is found i.e. parent_dir is not None
+                # then do the following:
+                #   - extract the scoring program
+                #   - move its content to destination dir
+                #   - delete the parent dir
+
+                # Get list of files and directories from the zip of scoring program
+                extracted_files = z.namelist()
+
+                # Intiialize parent dir with a None value
+                parent_dir = None
+
+                # loop over all the extracted files to identify parent directory
+                for extracted_file in extracted_files:
+                    # if a metadata file is located in the subdirectory
+                    # that directory is considered the parent dir
+                    # Note:
+                    # `/` shows that there is a directory structure e.g. scoring_program/metadata
+                    if '/' in extracted_file and os.path.basename(extracted_file) == 'metadata':
```

Comment:
@Didayolo here i have used condition of `metadata` but I think we can have different metadata files like metadata.yaml

We should discuss this

Reply:
Yes, you can see here that metadata file can be `metadata` or `metadata.yaml`:

https://github.com/codalab/codabench/blob/d2f1197b79b1e613214297f0f2404cea11c9251d/compute_worker/compute_worker.py#L527C9-L527C9

Reply:
> Yes, you can see here that metadata file can be `metadata` or `metadata.yaml`:
> 
> https://github.com/codalab/codabench/blob/d2f1197b79b1e613214297f0f2404cea11c9251d/compute_worker/compute_worker.py#L527C9-L527C9

Adding a condition of metadata.yaml